internal/run/studioHandlers.go

```diff
@@ -0,0 +1,126 @@
+package run
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/speakeasy-api/sdk-gen-config/workflow"
+	"github.com/speakeasy-api/speakeasy-core/errors"
+	"github.com/speakeasy-api/speakeasy/internal/run/generated-studio-sdk/models/components"
+	"github.com/speakeasy-api/speakeasy/internal/utils"
+)
+
+type StudioHandlers struct {
+	Workflow Workflow
+}
+
+func (h *StudioHandlers) health(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+
+	workflow, err := convertWorkflowToComponentsWorkflow(h.Workflow.workflow)
+	if err != nil {
+		return fmt.Errorf("error converting workflow to components.Workflow: %w", err)
+	}
+
+	ret := components.HealthResponse{
+		Workflow:         workflow,
+		TargetID:         h.Workflow.Target,
+		WorkingDirectory: os.Getenv("PWD"),
+	}
+	err = json.NewEncoder(w).Encode(ret)
+	if err != nil {
+		return fmt.Errorf("error encoding health response: %w", err)
+	}
+
+	return nil
+}
+
+func (h *StudioHandlers) getSource(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	var err error
+
+	workflow := h.Workflow
+
+	source, err := findWorkflowSourceBasedOnTarget(workflow, workflow.Target)
```

Comment:
perhaps throw before launching the studio as if we can't find a source the studio will be non-functional